projects/plugins/boost/app/class-jetpack-boost.php

```diff
@@ -159,6 +160,7 @@ private function register_deactivation_hook() {
 	 */
 	public function clear_cache() {
 		do_action( 'jetpack_boost_clear_cache' );
+		Speed_Score_Request::clear_cache();
```

Comment:
Where possible, we try to avoid code specific to individual features or modules out of the root class in Jetpack Boost. IMHO, we should have the `Speed_Score` class (instantiated inside `class Admin`) react to the `jetpack_boost_clear_cache` hook so the Jetpack Boost entry-level object can be kept as simple as possible.

I'm aware that is inconsistent with the Viewport and Critical CSS stuff that's crept into the root level object, but I view those as mess we should clean up. :)

Reply:
I have committed an update with your suggestion. One thing that struck me a bit, `Speed_Score` is mainly responsible for the rest endpoints. Clearing the cache seems to be a bit outside of it's responsibilities.

Reply:
@eAdnan007, I definitely see your last point but I am having difficulties to see where we can put the logic if we want to try keeping the Jetpack Boost entry-level object as simple as possible. Maybe with your change, it is an acceptable solution for now.

@thingalon: I have created this https://github.com/Automattic/jetpack/issues/20760 issue to refactor the Critical CSS dependencies in the Jetpack Boost class.

The Viewport dependencies can be reviewed when we start working on this existing https://github.com/Automattic/jetpack/issues/20562 issue

Reply:
Thanks; this looks much cleaner :) Thanks @davidlonjon for the cleanup GH issues too!